liftwing/models/revertrisk.py

```diff
@@ -1,28 +1,29 @@
-import requests
+from pydantic import BaseModel, Field, model_validator
 from .liftwing_model import LiftwingModel
+from ..utils.validator import validate_payload, check_required_fields
 from typing import Any, Dict
 
 
-class RevertRiskAPIModel(LiftwingModel):
-    def __init__(self, base_url="https://api.wikimedia.org/service/lw/inference/v1/models/revertrisk-language-agnostic:predict"):
-        super().__init__(base_url)
+class RevertRiskPayload(BaseModel):
+    lang: str = Field(...,
+                      title="Language code of the wiki",
+                      description='A string representing the language code related to the target '
+                                  'wiki. Example: "en" for English Wikipedia.')
+    rev_id: int = Field(...,
+                        title="Revision ID of the edit",
+                        description="The revision id for the wiki identified by the lang parameter.")
 
-    def request(self, payload: Dict[str, Any], method: str = "POST", headers: Dict[str, str] = None) -> Dict[str, Any]:
-        """
-        This function makes a POST request to https://api.wikimedia.org/service/lw/inference/v1/models/revertrisk-language-agnostic:predict
-        using the language parameter and returns a JSON
-        language is for the different wiki languages, rev_id is the specific revisions
-        """
-        language = payload.get("lang")
-        if language is None:
-            raise ValueError("'lang' parameter is required in the payload.")
-        rev_id = payload.get("rev_id")
-        if rev_id is None:
-            raise ValueError("rev_id is none, add revision id to continue")
+    @model_validator(mode='before')
+    def check_fields(cls, values):
+        check_required_fields(values, ['lang', 'rev_id'])
+        return values
 
-        if headers is None:
-            headers = {}
 
-        response = requests.post(self.base_url, json=payload, headers=headers)
+class RevertRiskAPIModel(LiftwingModel):
+    def __init__(self, base_url="https://api.wikimedia.org/service/lw/inference/v1/models/revertrisk-language-agnostic:predict"):
+        super().__init__(base_url)
 
-        return response.json()
+    @validate_payload(RevertRiskPayload)
+    def request(self, payload:  Dict[str, Any], method: str = "POST",
+                headers: Dict[str, str] = None) -> Dict[str, Any]:
+        return super().request(payload=payload, method=method, headers=headers)
```

Comment:
```suggestion
    def request(self, payload:  Dict[str, Any], method: str = "POST",
                headers: Dict[str, str] = None) -> Dict[str, Any]:
        _ = RevertRiskPayload(**payload)
        return super().request(payload=payload, method=method, headers=headers)
```
I wonder if the above might be a simpler approach. In the `validate_payload`, you do two things:
1) check if payload exists
but the payload is required for request() since it doesn't have a default value. This check may not be necessary.

2) pass payload to the pydantic model
This could be done by adding a single line in the request(), then we don't need the entire decorator.

I feel using a decorator is not very straightforward, but maybe it is easier for us to extend it in the future.

Reply:
The dummy step `_ = RevertRiskPayload(**payload)` (same as the one in validator.py `_ = model(**payload)`) is a bit weird. I'd prefer to hide that kind of complexity from the end user by using a decorator.
Then when we have instructions on how to add a new model, the user will just add the model and define the proper validation model in the decorator.
I'm trying to think of a way to even hide that complexity away in order to make adding a model even simpler but haven't figured out a proper solution yet.

Reply:
Yes, you have a point. It's better to hide the complexity from the end user. Let's stay as it is.